Content.Server/Speech/EntitySystems/FrenchAccentSystem.cs

```diff
@@ -0,0 +1,38 @@
+using Content.Server.Speech.Components;
+using System.Text.RegularExpressions;
+
+namespace Content.Server.Speech.EntitySystems;
+
+public sealed class FrenchAccentSystem : EntitySystem
+{
+    [Dependency] private readonly ReplacementAccentSystem _replacement = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<FrenchAccentComponent, AccentGetEvent>(OnAccentGet);
+    }
+
+    // converts left word when typed into the right word. For example typing you becomes ye.
+    public string Accentuate(string message, FrenchAccentComponent component)
+    {
+        var msg = message;
+
+        msg = _replacement.ApplyReplacements(msg, "french");
+
+        // replaces th with dz 
+        msg = Regex.Replace(msg, @"th", "dz", RegexOptions.IgnoreCase);
```

Comment:
In my experience, native French speakers do not tend to turn all "th" into "dz"; it tends to become "z" or "s" depending on whether it was voiced or not.

Suggestion: don't mess with "th" unless the logic is made much smarter.

Reply:
i changed it to 'z i think this is a better overall solution. I dont think accent needs to be exactly perfect with all logic cases since its typically aimed towards english speaking servers.